libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -341,6 +341,16 @@ class MapboxTripSession(
         }
     }
 
+    /**
+     * Start navigating a new leg in [DirectionsRoute]
+     *
+     * @param routeIndex the index of the route to navigate to
+     * @param legIndex the index of the leg to navigate to
+     */
+    override fun updateRouteLegIndex(routeIndex: Int, legIndex: Int) {
+        navigator.updateLegIndex(routeIndex, legIndex)
```

Comment:
See comment above re: `routeIndex`

Reply:
See comment above re: misusing `MapboxNavigation#navigateToNextStop`

Maybe we can rely on the response returned by the `Navigator` `NavigationStatus`

```kotlin
/**                                                                            
 * Follows a new route and leg of the already loaded directions.               
 * Returns an initialized route state if no errors occurred                    
 * otherwise, it returns an invalid route state.                               
 *                                                                             
 * @param routeIndex new route index                                           
 * @param legIndex new leg index                                               
 *                                                                             
 * @return an initialized route state as [NavigationStatus]                    
 */                                                                            
override fun updateLegIndex(routeIndex: Int, legIndex: Int): NavigationStatus =
    navigator.changeRouteLeg(routeIndex, legIndex)                             
```

Reply:
yeah, actually that may help me figure out why it's not working

Reply:
updated. also found that the navigator status is being updated correctly